packages/create-sitecore-jss/src/templates/nextjs-multisite/src/lib/sitemap-fetcher/plugins/graphql-sitemap-service.ts

```diff
@@ -0,0 +1,36 @@
+﻿import {
+  MultisiteGraphQLSitemapService,
+  StaticPath,
+  constants,
+  SiteInfo,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import config from 'temp/config';
+import { SitemapFetcherPlugin } from '..';
+import { GetStaticPathsContext } from 'next';
+import { siteResolver } from 'lib/site-resolver';
+
+class GraphqlSitemapServicePlugin implements SitemapFetcherPlugin {
+  _graphqlSitemapService: MultisiteGraphQLSitemapService;
+
+  constructor() {
+    this._graphqlSitemapService = new MultisiteGraphQLSitemapService({
+      endpoint: config.graphQLEndpoint,
+      apiKey: config.sitecoreApiKey,
+      sites: [...new Set(siteResolver.sites.map((site: SiteInfo) => site.name))],
+    });
+  }
+
+  async exec(context?: GetStaticPathsContext): Promise<StaticPath[]> {
+    if (process.env.EXPORT_MODE) {
+      if (process.env.JSS_MODE === constants.JSS_MODE.DISCONNECTED) {
+        return [];
+      }
+      return process.env.EXPORT_MODE
+        ? this._graphqlSitemapService.fetchExportSitemap(config.defaultLanguage)
+        : this._graphqlSitemapService.fetchSSGSitemap(context?.locales || []);
+    }
+    return this._graphqlSitemapService.fetchSSGSitemap(context?.locales || []);
```

Comment:
Some redundant logic here (and in `nextjs` template's version). Should be able to have just the following, right?

```suggestion
    if (process.env.JSS_MODE === constants.JSS_MODE.DISCONNECTED) {
      return [];
    }
    return process.env.EXPORT_MODE
      ? this._graphqlSitemapService.fetchExportSitemap(config.defaultLanguage)
      : this._graphqlSitemapService.fetchSSGSitemap(context?.locales || []);
```